refsheet.cls

```diff
@@ -1,57 +1,212 @@
+%
+\DeclareVoidOption{portrait}{%
+  \PackageWarning{refsheet}{Incompatible with portrait mode}}
+\DeclareVoidOption{twocolumn}{%
+  \PackageError{refsheet}{Using 'multicol' for columns}}
+\DeclareVoidOption{titlepage}{%
+  \PackageWarning{refsheet}{Incompatible with a title page}}
+%
+%   Pass all other options to article
+%
+\DeclareDefaultOption{%
+  \PassOptionsToClass{\CurrentOptionKey}{article}}
+%
+%   DEV: Space for more options
+%
+
+%
+%   Process these options
+%   
+\ProcessKeyvalOptions{refsheet}
+%
+%%%
+
+%%% INHERIT STANDARD CLASS
+%
+%   Latest update of standard class article
+%   (It probably also works with earlier versions, but why sweat it.)
+\LoadClass{article}[2014/09/29]
+%
+%%%
+
+%%% ADDITIONAL REQUIRED TOOLS
+%
+%   Latest update of the nag package. 
+%   Harasses the user about deprecated usages.
+%
+\RequirePackage[orthodox,l2tabu,abort]{nag}[2011/11/25] 
+%
+%   Package etoolbox is necessary for additional hooks like \AtEndPreamble
+%
+\RequirePackage{etoolbox}
+%
+%   Page layout with latest version of geometry
+%
+\RequirePackage{geometry}[2018/04/16]
+\AtEndPreamble{%
+  \geometry{landscape,margin=\refsheet@margin}}
+%
+%   Produce columns with multicol
+%
+\RequirePackage{multicol}[2018/04/20]
+\AfterEndPreamble{%
+  \begin{multicols}{\refsheet@rscols}}
+\AtEndDocument{%
+  \end{multicols}}
+%
+%%%
+
+%%% REDEFINE TITLE
+%
+\renewcommand{\maketitle}{%
+  {%
+   \begin{center}%
+     \Large \@title \\%
+     \vspace{0.1ex}%
+     \small \@author, \@date\end{center}%
+  }%
+}
+%
+%%%
 
-\ProcessOptions
-
-\LoadClass{article}[2013/02/25]
-
-\RequirePackage{fixltx2e}
-\RequirePackageWithOptions{nag}[2013/03/25 othodox,l2tabu,abort]
-
-%% Page layout
-
-\RequirePackage{geometry}[2013/03/25]
-%\usepackage[landscape,margin=0.5in]{geometry}
-\geometry{landscape,margin=0.5in}
-\RequirePackage{multicol}
-
+%%% DOCUMENT DIVISIONS
+%
+%   Package titlesec for setting spacing
+%
+\RequirePackage{titlesec}
+%
+%   Package nopageno so there are no page numbers
+%   (to keep \section from resetting page style).
+%
+\RequirePackage{nopageno} 
+%
+%   No section numbering
+%
+\setcounter{secnumdepth}{0}
+%
+%   No indentation and superfluous spacing
+%
+\setlength{\parindent}{0pt} 
+\titlespacing{\section}{0pt}{0pt}{0pt}
+\titlespacing{\subsection}{0pt}{0pt}{0pt}
+%
+%%%
+
+%%% COMMANDS AND ENVIRONMENTS
+%
+%   Extended definition of lists with enumitem
+%
+\RequirePackage[inline]{enumitem}[2011/08/28]
+%   Calculating widths, e.g. with \widthof
+\RequirePackage{calc} 
+
+% generate dense lists
+\setlist{noitemsep,topsep=0.05ex,}
+\setlist[itemize]{label=\textendash,leftmargin=*}
+\setlist[description]{leftmargin=!}
+% Provide a list-type for truetype descriptions
+% This is retained for backward compatibility
+\newlist{ttdesc}{description}{1}
+\setlist[ttdesc]{font=\ttfamily}
+% Conveniently assigning the longest label as 
+\newenvironment{rslisttt}[1]
```

Comment:
You can abstract this by pulling the `\ttfamily` or `\bfseries` tokens out into a macro variable if you wanted.

Reply:
Originally I wanted to stay as close to the example as possible and necessary. Later I noticed it would be nice to have bold descriptions, etc., so I added them. When I finished writing the class I had the same idea, but wanted to start using the class instead of developing it more. It works, even though it's not overly elegant.

You can see that I used this approach for the inline description lists ([L172](https://github.com/polyluxus/refsheet/blob/91e46be863259fd87d05e6be77f966e263d3190f/refsheet.cls#L172)).

I might come back to that, but for the time being, I decided to stick with it.